service/src/main/kotlin/app/cash/backfila/service/runner/BackfillRunner.kt

```diff
@@ -108,27 +108,27 @@ class BackfillRunner private constructor(
   }
 
   @VisibleForTesting
-  fun start(coroutineScope: CoroutineScope) {
+  fun CoroutineScope.start() {
```

Comment:
I forget why I couldn't make this work. Seems good but wrong PR?

Reply:
Yeah it turns out all of this is wrong according to @bezmax and @keeferrourke . I'll work with them in a followup to fix this more. Apparently we are blocking when we shouldn't on the main threads. There is also this really cool Flows thing that should work really really well.

Reply:
lets talk through that before big changes